packages/devtools_app/lib/src/debugger/flutter/console.dart

```diff
@@ -17,20 +18,144 @@ import 'debugger_controller.dart';
 // TODO(devoncarew): Support hyperlinking to stack traces.
 
 /// Display the stdout and stderr output from the process under debug.
-class Console extends StatefulWidget {
-  const Console({
+class DebuggerConsole extends StatefulWidget {
+  const DebuggerConsole({
     Key key,
     this.controller,
   }) : super(key: key);
 
   final DebuggerController controller;
 
   @override
-  _ConsoleState createState() => _ConsoleState();
+  _DebuggerConsoleState createState() => _DebuggerConsoleState();
+
+  static const copyToClipboardButtonKey =
+      Key('debugger_console_copy_to_clipboard_button');
+  static const clearStdioButtonKey = Key('debugger_console_clear_stdio_button');
 }
 
-class _ConsoleState extends State<Console> {
-  final scrollController = ScrollController();
+class _DebuggerConsoleState extends State<DebuggerConsole> {
+  var _lines = <String>[];
+
+  void _onStdioChanged() {
+    setState(() {
+      _lines = widget.controller.stdio.value;
+    });
+  }
+
+  @override
+  void initState() {
+    _lines = widget.controller.stdio.value;
+    widget.controller.stdio.addListener(_onStdioChanged);
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    widget.controller.stdio.removeListener(_onStdioChanged);
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Console(
+      lines: _lines,
+      controls: [
+        if (_lines.isNotEmpty) ...[
+          IconButton(
```

Comment:
add a TODO to pull these action widgets into common_widgets.dart if they will also be used in the logging screen.

Reply:
Thanks for the super fast review!

I'll add a TODO there. I'll get back to the team to find the best place to put the extracted code next week!